scale-tests/deploy-dispatchers.py

```diff
@@ -167,30 +281,75 @@ def get_default_options(arguments: dict) -> dict:
     return options
 
 
-if __name__ == "__main__":
-    arguments = docopt(__doc__, version="deploy-dispatchers.py 0.0")
+def get_quota_options(arguments: typing.Dict) -> typing.Dict:
+    """
+    Move the quota options from the command line arguments to a dict.
+    """
+    create_quotas = ast.literal_eval(arguments.get("--create-quotas", True))
+    if not create_quotas:
+        return {}
+
+    resources = ["cpus", "mem", "gpus", ]
+    targets = ["drivers", "executors", ]
+
+    quota_options = {}
+    for t in targets:
+        quota_options[t] = {}
+        for r in resources:
+            arg_key = "--quota-{}-{}".format(t, r)
+            if arg_key in arguments:
+                quota_options[t][r] = arguments[arg_key]
+
+    return quota_options
 
-    options_file = arguments['--options-json']
+
+def install(args):
+    options_file = args['--options-json']
     if options_file:
         if not os.path.isfile(options_file):
             # TODO: Replace with logging
-            print("The specified file does not exist: %s", options_file)
+            log.error("The specified file does not exist: %s", options_file)
             sys.exit(1)
 
         options = json.load(open(options_file, 'r'))
     else:
-        options = get_default_options(arguments)
+        options = get_default_options(args)
+
+    if args['--package-repo']:
+        sdk_repository.add_stub_universe_urls([args['--package-repo']])
+
+    rc, _, _ = sdk_cmd.run_raw_cli("package install spark --cli --yes")
+    assert rc == 0, "Error installing spark CLI"
+
+    quota_options = get_quota_options(args)
 
-    deploy_dispatchers(
-        num_dispatchers=int(arguments['<num_dispatchers>']),
-        service_name_base=arguments['<service_name_base>'],
-        output_file=arguments['<output_file>'],
+    services = {}
+
+    services["spark"] = deploy_dispatchers(
+        num_dispatchers=int(args['<num_dispatchers>']),
+        service_name_base=args['<service_name_base>'],
+        output_file=args['<output_file>'],
+        linux_user=args["--user"],
         options=options,
-        package_repo=arguments['--package-repo'],
-        create_quotas=ast.literal_eval(arguments.get("--create-quotas", True)),
-        quota_drivers_cpus=arguments['--quota-drivers-cpus'],
-        quota_drivers_gpus=arguments['--quota-drivers-gpus'],
-        quota_drivers_mem=arguments['--quota-drivers-mem'],
-        quota_executors_cpus=arguments['--quota-executors-cpus'],
-        quota_executors_gpus=arguments['--quota-executors-gpus'],
-        quota_executors_mem=arguments['--quota-executors-mem'])
+        quota_options=quota_options)
+
+    output_filename = "{}-infrastrcuture.json".format(args["<output_file>"])
```

Comment:
Typo "infrastrcuture".

Reply:
e.g.
```bash
TEST_NAME=streaming
streaming-infrastructure.json
streaming-dispatchers.json
streaming-submissions.out
```

Reply:
Re: the name, keep in mind that batch and GPU tests would likely also read this file.